test/stdlib/TestData.swift

```diff
@@ -1183,6 +1183,57 @@ class TestData : TestDataSuper {
         expectEqual(Data(bytes: [1]), slice)
     }
 
+    // This test uses `repeatElement` to produce a sequence -- the produced sequence reports its actual count as its `.underestimatedCount`.
+    func test_appendingNonContiguousSequence_exactCount() {
+        var d = Data()
+
+        // d should go from .empty representation to .inline.
+        // Appending a small enough sequence to fit in linline should actually be copied.
+        d.append(contentsOf: repeatElement(UInt8(0x01), count: 2))
```

Comment:
This could than become:
````swift
d.append(contentsOf: UInt8(0)...UInt8(1))
expectEqual(Data([0, 1]), d)

d.append(contentsOf: UInt8(2)...UInt8(2))
expectEqual(Data([0, 1, 2]), d)

d.append(contentsOf: UInt8(3)...UInt8(26))
expectEqual(Data([
  00, 01, 02, 03, 04, 05, 06, 07, 08, 09,
  10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 
  20, 21, 22, 23, 24, 25, 26]), d)
````